packages/task/src/node/task-server.ts

```diff
@@ -112,6 +126,28 @@ export class TaskServerImpl implements TaskServer, Disposable {
                             problems
                         });
                     }
+                    if (taskConfiguration.isBackground) {
+                        const backgroundTaskStatus = this.backgroundTaskStatusMap.get(event.taskId);
+
+                        const isBeginsPatternMatch = collector.matchBeginMatcher(event.line);
```

Comment:
please correct me if I am wrong: we don't need to call `matchBeginMatcher()` if `backgroundTaskStatus!.isBeginsPatternMatch === true || activeOnStart === true`, do we?

Reply:
fixed, done